scripting/lua/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+# scripting/lua/CMakeLists.txt
+# Lua scripting build configuration
+
+# ---------------------- Declare lua scripting library options -----------------------
+
+option(CUBOS_SCRIPTING_LUA "Build lua scripting library" ON)
+
+message("# Building cubos::scripting::lua" ${CUBOS_SCRIPTING_LUA})
+
+# ----------------------- Define lua scripting source files -----------------------
+
+set(SCRIPTING_LUA_SOURCE
+    "src/plugin.cpp"
+)
+
+# ------------------------ Configure lua scripting target -------------------------
+
+add_library(cubos-scripting-lua ${SCRIPTING_LUA_SOURCE})
+target_link_libraries(cubos-scripting-lua PRIVATE cubos::engine cubos-api-core cubos-api-engine)
+target_include_directories(cubos-scripting-lua PUBLIC "include")
+cubos_common_target_options(cubos-scripting-lua)
+
+# ------------- Fetch and link private library dependencies -------------
+
+include(FetchContent)
+
+# Fetch stb_image
+FetchContent_Declare(
+	lua
+	GIT_REPOSITORY "https://github.com/GameDevTecnico/cubos-lua.git"
+    GIT_TAG "origin/main"
```

Comment:
Ptobably a better idea to refer the specific commit hash. Otherwise, old cubos versions might break when we update the lua repo